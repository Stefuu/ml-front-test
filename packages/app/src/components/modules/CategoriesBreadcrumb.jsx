import React from 'react'
import Box from '../elements/Box'
import styled from 'styled-components'
import { rem } from 'polished'
import { theme } from 'styled-tools'

const Container = styled(Box)`
 color: ${theme('colors.b40')};
 font-size: 14px;
 span {
   :last-child {
     font-weight: bold;
   }
 }
`

const CategoriesBreadcrumb = (props) => (
  <Container
    py={rem(16)}>
    {props.categories.map((cat, index) => {
      const lastIndex = props.categories.length !== index + 1
      return (
        <span key={index}>
          {cat}{lastIndex ? ' > ' : ''}
        </span>
      )
    })}
  </Container>
)

export default CategoriesBreadcrumb
