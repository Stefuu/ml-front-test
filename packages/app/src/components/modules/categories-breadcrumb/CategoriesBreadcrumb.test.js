import React from 'react'
import { render } from 'react-testing-library'
import CategoriesBreadcrumb from './CategoriesBreadcrumb'

describe('CategoriesBreadcrumb', () => {
  it('should render [snapshot]', () => {
    const { container } = render(<CategoriesBreadcrumb categories={['Cat, Dog, Bird']} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
