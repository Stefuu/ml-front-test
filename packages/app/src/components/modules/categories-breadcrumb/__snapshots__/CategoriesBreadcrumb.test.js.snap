// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesBreadcrumb should render [snapshot] 1`] = `
<div
  class="sc-bdVaJa sc-bwzfXH gwEKIb"
>
  <span>
    Cat, Dog, Bird
    
  </span>
</div>
`;
